baremetal/metal3remediation_manager_test.go

```diff
@@ -345,6 +346,190 @@ var _ = Describe("Metal3Remediation manager", func() {
+				Expect(patchError).NotTo(BeNil())
+			}
+		},
+		Entry("Should set the unhealthy annotation", testCaseSetUnhealthyAnnotation{
+			M3Machine: &capm3.Metal3Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            "mym3machine",
+					Namespace:       "myns",
+					OwnerReferences: []metav1.OwnerReference{},
+					Annotations: map[string]string{
+						HostAnnotation: "myns/myhost",
+					},
+				},
+			},
+			Host: &bmh.BareMetalHost{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "myhost",
+					Namespace:   "myns",
+					Annotations: map[string]string{capm3.UnhealthyAnnotation: ""},
+				},
+			},
+			ExpectTrue: true,
+		}),
+		Entry("Should not set the unhealthy annotation, annotation is empty", testCaseSetUnhealthyAnnotation{
+			M3Machine: &capm3.Metal3Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            "mym3machine",
+					Namespace:       "myns",
+					OwnerReferences: []metav1.OwnerReference{},
+					Annotations:     map[string]string{},
+				},
+			},
+			Host: &bmh.BareMetalHost{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "myhost",
+					Namespace:   "myns",
+					Annotations: map[string]string{capm3.UnhealthyAnnotation: ""},
+				},
+			},
+			ExpectTrue: false,
+		}),
+		Entry("Should not set the unhealthy annotation because of wrong HostAnnotation", testCaseSetUnhealthyAnnotation{
+			M3Machine: &capm3.Metal3Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            "mym3machine",
+					Namespace:       "myns",
+					OwnerReferences: []metav1.OwnerReference{},
+					Annotations: map[string]string{
+						HostAnnotation: "myns/wronghostname",
+					},
+				},
+			},
+			Host: &bmh.BareMetalHost{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "myhost",
+					Namespace:   "myns",
+					Annotations: map[string]string{capm3.UnhealthyAnnotation: ""},
+				},
+			},
+			ExpectTrue: false,
+		}),
+	)
+
+	DescribeTable("Test SetRebootAnnotation",
+		func(tc testCaseSetUnhealthyAnnotation) {
+			fakeClient := fake.NewClientBuilder().WithScheme(setupScheme()).WithObjects(tc.Host).Build()
+			remediationMgr, err := NewRemediationManager(fakeClient, nil, tc.M3Machine, nil,
+				logr.Discard(),
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			patchError := remediationMgr.SetRebootAnnotation(context.TODO())
+
+			if tc.ExpectTrue {
+				Expect(patchError).To(BeNil())
+
+			} else {
+				Expect(patchError).NotTo(BeNil())
+			}
+		},
+		Entry("Should set the reboot annotation", testCaseSetUnhealthyAnnotation{
+			M3Machine: &capm3.Metal3Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            "mym3machine",
+					Namespace:       "myns",
+					OwnerReferences: []metav1.OwnerReference{},
+					Annotations: map[string]string{
+						HostAnnotation: "myns/myhost",
+					},
+				},
+			},
+			Host: &bmh.BareMetalHost{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "myhost",
+					Namespace:   "myns",
+					Annotations: map[string]string{capm3.UnhealthyAnnotation: ""},
```

Comment:
Why are we setting UnhealthyAnnotation when test entry is all about reboot annotation?